Detections/ASimWebSession/PossibleDGAContacts.yaml

```diff
@@ -0,0 +1,104 @@
+id: 9176b18f-a946-42c6-a2f6-0f6d17cd6a8a
+name: Possible contact with a domain generated by a DGA
+description: |
+  'Identifies contacts with domains names that might have been generated by a Domain Generation Algorithm (DGA). DGAs can be used
+  by malware to generate rendezvous points that are difficult to predict in advance. This detection uses the Alexa Top 1 million domain names to build a model
+  of what normal domains look like. It uses this to identify domains that may have been randomly generated by an algorithm.
+  The triThreshold is set to 500 - increase this to report on domains that are less likely to have been randomly generated, decrease it for more likely.
+  The start time and end time look back over 6 hours of data and the dgaLengthThreshold is set to 8 - meaning domains whose length is 8 or more are reported.'
+severity: Medium
+requiredDataConnectors: []
+queryFrequency: 6h
+queryPeriod: 6h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CommandAndControl
+tags:
+  - Id: 4acd3a04-2fad-4efc-8a4b-51476594cec4
+    version: 1.0.0
+  - Schema: ASIMNetworkSession
+    SchemaVersion: 0.2.0
+relevantTechniques:
+  - T1568
+query: |
+    let triThreshold = 500;
+    let querystarttime = 6h;
+    let dgaLengthThreshold = 8;
+    // fetch the alexa top 1M domains
+    let top1M =  (externaldata (Position:int, Domain:string)   [@"http://s3-us-west-1.amazonaws.com/umbrella-static/top-1m.csv.zip"]  with (format="csv", zipPattern="*.csv"));
+    // extract tri grams that are above our threshold - i.e. are common
+    let triBaseline =   top1M
+      | extend Domain = tolower(extract("([^.]*).{0,7}$", 1, Domain))
+      | extend AllTriGrams = array_concat(extract_all("(...)", Domain), extract_all("(...)", substring(Domain, 1)), extract_all("(...)", substring(Domain, 2)))
+      | mvexpand Trigram=AllTriGrams to typeof(string)
+      | summarize triCount=count() by Trigram
+      | sort by triCount desc
+      | where triCount > triThreshold
+      | distinct Trigram;
+    // collect domain information from common security log, filter and extract the DGA candidate and its trigrams
+    let allDataSummarized =  imWebSession(starttime=querystarttime)  // CommonSecurityLog
```

Comment:
- Why? the time window is anyways 6 hours.
- The original is actually not about URL but about RemoteHostName. We should probably have one detection on websession using url and another on networksession using domain. It does raise an issue: we probalby want the imNetworkSession to be mutually exclusive with imWebSession. What do you think?

Reply:
Maybe union is preferable.

Reply:
Time: agreed.
The "RemoteHost" in the original is the host part of the url, this is why they are "trimming" the TLD. 
I dont think there's any point in adding NW sessions. The generation will be of an internet domain, not of an internal host.
Dns... yes. we can add a detection at DNS level